content/matplotlib/concepts/pyplot/terms/hist/hist.md

```diff
@@ -0,0 +1,82 @@
+---
+Title: '.hist()'
+Description: 'Creates a histogram plot that illustrates the frequency of values within a given range.'
+Subjects:
+  - 'Data Science'
+  - 'Data Visualization'
+Tags:
+  - 'Graphs'
+  - 'Libraries'
+  - 'Matplotlib'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/data-science'
+---
+
+The **`.hist()`** method in the matplotlib library is used to draw a histogram plot, showing the frequency of values within a given range.
+
+## Syntax
+
+```pseudo
+matplotlib.pyplot.hist(x, bins, range, density, weights, cumulative, bottom, histtype, align, orientation, rwidth, log, color, label, stacked)
+```
+
+The `x` argument is the only required parameter. It represents the values that will be plotted and can be of type float or array.
+
+Other parameters are optional and can be used to customize plot elements such as the number of bins, color, and other relevant features.
+
+`.hist()` takes the following arguments:
+
+- `x`: Values to be plotted of type float or array.
+- `bins`: Sets the number of bins, otherwise the number is autogenerated by default.
+- `range`: The range of values to consider as a tuple in lieu of the entire range included in the data.
+- `density`: Adds a probability density line (overlay) to the plot, default is `False`.
+- `weights`: Accepts an array of values (must match the shape of `x`) that determines the relative contribution of each value in `x`.
+- `cumulative`: Takes a boolean value, and if `True` computes a cumulative histogram where each bin gives counts of the relevant values within that bin in addition to all the values smaller than that bin.
+- `bottom`: Sets the floor or the bottom of the bins, default is 0.
+- `histtype`: Takes keywords (e.g. `bar`, `barstacked`, `step`, `stepfilled`) to modify the histogram type.
+- `align`: Sets the alignment of the bins, default is `mid` (options include `left` and `right`).
+- `orientation`: Sets the orientation of the histogram, default is `vertical`.
+- `rwidth`: Sets the bar width as a fraction of the bin size.
+- `log`: Changes the "y" values, the frequency of values, to a log scale.
+- `color`: Sets the color of the lines/bars.
+- `label`: Option to include names for multiple data sets.
+- `stacked`: Boolean value that transforms the plot to be stacked, similar to a stacked bar graph.
+
+## Example
+
+The example below demonstrates the use of `.hist()` to plot a histogram figure.
+
+```py
+import pandas as pd
+import matplotlib.pyplot as plt
+import math
+
+# Data is available at https://data.austintexas.gov/Health-and-Community-Services/Austin-Animal-Center-Intakes/wter-evkm
+df = pd.read_csv('/content/Austin_Animal_Center_Intakes.csv')
+
+# Function for standardizing pet age values
+def get_months(x):
+    if x.endswith('years'):
+        return abs(int(x.split(' ')[0]) * 12) # Some year entries included negative values
+    elif x.endswith('weeks'):
+        return math.ceil(int(x.split(' ')[0]) / 4 )
+    else:
+        return int(x.split(' ')[0])
```

Comment:
Would it be possible to set up two variables here? One for number (the int that's extracted) and another for the unit? In addition, are there some strings with no units?

Reply:
This is supposed to be a quick and dirty function to create a new column, to "explore" the data. It's not really meant as a function for reuse. The data being addressed is a bit of a mess, it includes three different units, negative values and NaNs. I did update the last line because I hadn't noticed the "days" entries (so I'm rounding any such entry up to 1 month).

Reply:
I see, did we want to keep an `elif` for the case of month(s)? I think this is good to go after

Reply:
I bungled the edit. When I reviewed the data again I also saw that some values are not plural (1 month vs 3 months), so I redesigned the function again. Should be all set.